engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -575,10 +581,20 @@ protected void serviceInit(Configuration conf) throws Exception
       this.appDataPushAgent = new AppDataPushAgent(dnmgr, appContext);
       addService(this.appDataPushAgent);
     }
-    // initialize all services added above
+    initApexPluginDispatcher();
+
+    // Initialize all services added above
     super.serviceInit(conf);
   }
 
+  private void initApexPluginDispatcher()
+  {
+    PluginLocator locator = new ServiceLoaderBasedPluginLocator();
```

Comment:
How about going with the same approach to load plugins as the dag setup plugins by using a property that specifies the list of classes.